example-todomvc/src/main/scala/mhtml/todo/MhtmlTodo.scala

```diff
@@ -0,0 +1,342 @@
+      <footer class="footer" style:display={display}>
+        <ul class="filters">{todoLists.map(todoListsFooter)}</ul>
+        <button onclick={() =>
+            allTodos.map(_.filter(_.completed).foreach(todo =>
+              removeTodos.update(rtList => todo :: rtList)
+            ))
+            ()
+          }
+          class="clear-completed"
+          style:visibility={visibility}>
+          Clear completed
+        </button>
+      </footer>
+    Component(footerNode, removeTodos)
+  }
+
+  case class TodoUpdate(oldTodo: Todo, newTodo: Todo)
+  case class TodoListItemData(removal: Option[Todo], update: Option[TodoUpdate])
+  //
+  def todoListItem(todo: Todo): Component[TodoListItemData] = {
+    val removeTodo = Var[Option[Todo]](None)
+    val updateTodo = Var[Option[TodoUpdate]](None)
+    val suppressOnBlur = Var(false)
+    def submit: Event => Unit = { event: Event =>
+      suppressOnBlur := true
+      editingTodo := None
+      event.currentTarget.asInstanceOf[HTMLInputElement].value.trim match {
+        case "" =>
+          removeTodo := Some(todo)
+        case trimmedTitle =>
+          updateTodo := Some(TodoUpdate(todo, Todo(trimmedTitle, todo.completed)))
+      }
+    }
+    def onEditTodoTitle = { event: KeyboardEvent =>
+      event.keyCode match {
+        case KeyCode.Escape =>
+          suppressOnBlur := true
+          editingTodo := None
+        case KeyCode.Enter =>
+          submit(event)
+          focusInput()
+        case _ =>
+      }
+    }
+    def ignoreEvent: (Event) => Unit = _ => Unit
+    def blurHandler: Rx[Event => Unit] =
+      suppressOnBlur.map(x => if (x) ignoreEvent else submit)
+    def onToggleCompleted: (Event) => Unit = { event: Event =>
+      event.currentTarget match {
+        case input: HTMLInputElement =>
+          updateTodo := Some(TodoUpdate(todo, Todo(todo.title, input.checked)))
+        case _ =>
+      }
+    }
+    val onDoubleClick: (Event) => Unit = { _: Event =>
+      editingTodo := Some(todo)
+      focusInput()
+    }
+    val onDelete: (Event) => Unit = _ =>
+      removeTodo := Some(todo)
+
+    val css = editingTodo.map { x =>
+      val editing = if (x.contains(todo)) "editing" else ""
+      val completed = if (todo.completed) "completed" else ""
+      s"$editing $completed"
+    }
+    val data = removeTodo.map{
+      rmTodo => TodoListItemData(rmTodo, None)
+    }
+
+    val todoListElem =
+      <li class={css}>
+        <div class="view">
+          <input onclick={onToggleCompleted}
+                 class="toggle"
+                 type="checkbox"
+                 checked={conditionalAttribute(todo.completed)} />
+          <label ondblclick={onDoubleClick}>{ todo.title }</label>
+          <button onclick={onDelete} class="destroy"></button>
+        </div>
+        <input onkeydown={onEditTodoTitle}
+               id="editInput"
+               class="edit"
+               value={todo.title}
+               onblur={blurHandler}/>
```

Comment:
First time I see a `Rx[Event => Unit]` attribute in use, I'm not even sure we have a test for that :smile: